packages/editor/src/components/provider/use-hide-bocks-from-inserter.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { addFilter, removeFilter } from '@wordpress/hooks';
+
+// These post types are "structural" block lists.
+// We should be allowed to use
+// the post content and template parts blocks within them.
+const POST_TYPES_ALLOWING_POST_CONTENT_TEMPLATE_PART = [
+	'wp_block',
+	'wp_template',
+	'wp_template_part',
+];
+
+/**
+ * In some specific contexts,
+ * the template part and post content blocks need to be hidden.
+ *
+ * @param {string} postType Post Type
+ */
+export function useHideBlocksFromInserter( postType ) {
+	useEffect( () => {
+		/*
+		 * Prevent adding template part in the post editor.
```

Comment:
```suggestion
		 * Prevent adding template part in the editor.
```

Nit: Now this prevention is across both the post and site editors, is it worth making this comment a little more generic and not reference the post editor specifically?

Same for the other filter comment below.